### PR TITLE
chore: Speed up port-forward tests

### PR DESCRIPTION
**Pull request by mafredri**
I've been looking at some CI test timeout test failures, and whilst I haven't gotten to the bottom of it, I thought we could significantly increase the speed of the `port-forward` tests. On my machine the test time went from ~30s to ~5s, should be a bigger gain in CI.

- chore: Speed up port-forward tests
- chore: Add t.Helper and ensure listener closure on error

### CODE REVIEW

File: cli/portforward_test.go [hunk 1]
```diff
@@ -142,21 +142,22 @@ func TestPortForward(t *testing.T) {
 		},
 	}
 
+	// Setup agent once to be shared between test-cases (avoid expensive
+	// non-parallel setup).
+	var (
+		client       = coderdtest.New(t, &coderdtest.Options{IncludeProvisionerD: true})
+		user         = coderdtest.CreateFirstUser(t, client)
+		_, workspace = runAgent(t, client, user.UserID)
+	)
```
Comment (mafredri) [inline cli/portforward_test.go:151]: Note this change: We now share coderd/agent between tests. I didn't see any reason not-to.

**Review by Emyrk — CHANGES_REQUESTED** [review]
Smart!

**Review by Emyrk — APPROVED** [review]
Mean't to approve.

**Review by johnstcn — APPROVED** [review]
Nice!

File: cli/portforward_test.go (hunk 1 above)
Comment (mafredri) [reply cli/portforward_test.go:151]: That's an interesting point I did not think of. We'll also be sharing the parent `*testing.T` 😮. To be fair, with regards to test logging, I rarely find them helpful due to the verbosity / parallel execution. 😂

**Pull request merged**